src/core.ts

```diff
@@ -103,6 +105,8 @@ export const $ = new Proxy<Shell & Options>(
       let s
       if (Array.isArray(args[i])) {
         s = args[i].map((x: any) => $.quote(substitute(x))).join(' ')
+      } else if (is$Object(args[i])) {
+        s = dargs(args[i]).map((x: any) => $.quote(substitute(x))).join(' ')
```

Comment:
This doesn't support customizing the options for dargs.

Depending on the shell tool someone is calling from zx, I can see someone wanting to set [dargs options like `allowCamelCase`](https://github.com/sindresorhus/dargs#api), but this API via xz wouldn't support configuring it.

Also dargs doesn't seem to have an option for long opts with a single `-` (generating strings like `foo -bar=baz`) which some shell commands use. Maybe they would be open to adding that as an additional option though.